server-base/src/main/java/org/apache/kylin/rest/job/MetadataCleanupJob.java

```diff
@@ -122,7 +122,7 @@ public MetadataCleanupJob(KylinConfig config) {
 
         // exclude resources in use
         Set<String> activeResources = Sets.newHashSet();
-        for (CubeInstance cube : cubeManager.listAllCubes()) {
+        for (CubeInstance cube : cubeManager.reloadAndListAllCubes()) {
```

Comment:
In fact here when getting the `toDeleteCandidates`, it filters at candidates who were generated before NEW_RESOURCE_THRESHOLD_MS (12 hours). I think don't need to reload cube here. But it's OK

Reply:
In extremely case, NEW_RESOURCE_THRESHOLD_MS (12 hours) may be not enough. So it's better to keep it.